src/libraries/Directory.Build.targets

```diff
@@ -182,8 +182,13 @@
     <ItemGroup>
       <ProjectReference>
         <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
-        <AdditionalProperties Condition="'%(Filename)' == 'System.Private.CoreLib' and '$(RuntimeFlavor)' == 'CoreCLR'">Configuration=$(CoreCLRConfiguration);TargetFramework=$(NetCoreAppCurrent)</AdditionalProperties>
-        <AdditionalProperties Condition="'%(Filename)' == 'System.Private.CoreLib' and '$(RuntimeFlavor)' == 'Mono'">Configuration=$(MonoConfiguration);TargetFramework=$(NetCoreAppCurrent)</AdditionalProperties>
+        <UndefineProperties Condition="'%(Filename)' == 'System.Private.CoreLib'">$(UndefineProperties);TargetFramework</UndefineProperties>
+        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
+                                     '$(RuntimeFlavor)' == 'CoreCLR' and
+                                     '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</SetConfiguration>
+        <SetConfiguration Condition="'%(Filename)' == 'System.Private.CoreLib' and
```

Comment:
I suspect we might have slightly better perf by refactoring this as follows to avoid the number of statements which require batching.  But @rainersigwald can tell me if I'm right or not ;)
```xml
<ItemGroup>
  <ProjectReference>
    <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
  </ProjectReference>
  <ProjectReference Condition="'%(Filename)' == 'System.Private.CoreLib'">
    <SetConfiguration Condition="'$(RuntimeFlavor)' == 'CoreCLR' and
                                 '$(Configuration)' != '$(CoreCLRConfiguration)'">Configuration=$(CoreCLRConfiguration)</SetConfiguration>
    <SetConfiguration Condition="'$(RuntimeFlavor)' == 'Mono' and
                                 '$(Configuration)' != '$(MonoConfiguration)'">Configuration=$(MonoConfiguration)</SetConfiguration>
  </ProjectReference>
</ItemGroup>
```

Reply:
Sure I considered it but thought it wasn't worth it. Let me change that.